library/src/main/java/za/ampfarisaho/pathfinder/PathfinderNavDisplay.kt

```diff
@@ -44,7 +45,11 @@ fun PathFinderNavDisplay(
 
     val dialog by navigator.dialog.collectAsStateWithLifecycle()
     val backStack = rememberNavBackStack(*elements)
-    navigator.setBackStack(backStack)
+
+    DisposableEffect(backStack) {
```

Comment:
(me as i realize `navigator` should also be among the keys, and not just `backStack` 🥺)